core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -167,6 +172,54 @@ object StorageTool extends Logging {
     }
   }
 
+  def runFeatureDependenciesCommand(
+    namespace: Namespace,
+    printStream: PrintStream
+  ): Unit = {
+    val featureArg = namespace.getString("feature")
+    val Array(featureName, versionStr) = featureArg.split("=")
+
+    val featureLevel = try {
+      versionStr.toShort
+    } catch {
+      case _: NumberFormatException =>
+        throw new TerseFailure(s"Invalid version format: $versionStr for feature $featureName")
+    }
+
+    if (featureName == MetadataVersion.FEATURE_NAME) {
+      val metadataVersion = try {
+        MetadataVersion.fromFeatureLevel(featureLevel)
+      } catch {
+        case _: IllegalArgumentException =>
+          throw new TerseFailure(s"Unsupported metadata.version $featureLevel")
+      }
+      printStream.print(f"$featureName%s=$featureLevel%d (${metadataVersion.version()}%s) has no dependencies.%n")
+    } else {
+      Features.values().find(_.featureName == featureName) match {
+        case Some(feature) =>
+          val featureVersion = feature.fromFeatureLevel(featureLevel, true)
```

Comment:
do we have a catch if this fails?

Reply:
oh my no I missed that, i'll add it thanks!